tsdb/engine/tsm1/engine.go

```diff
@@ -1161,6 +1161,12 @@ func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions, aggregate bo
 				return err
 			}
 
+			if opt.Limit > 0 || opt.Offset > 0 || opt.SLimit > 0 || opt.SOffset > 0 {
```

Comment:
I reverted the sort removal so it will always sort as before.  It seems like it is unnecessary in some cases though.  For example,

```
SELECT SUM(value) FROM cpu GROUP BY host
```

Should not need to sort.  Is there a way to detect this scenario here?